### ISSUE TITLE
Add ability to mark tasks as complete

### ISSUE THREAD

**Issue by barricade823** [joshpalan/Tasky#15]
Completing a task would grey out the task but not make it disappear.

Idk about you but I hate when the things I've already done are hidden by default. I need reaffirmation about how cool am I and all the cool shit I've already finished. Maybe a hide/show all but still keep the greyed out.

**Comment by joshpalan** [comment]
I totally agree. We just gotta make sure we show it off in a fun way that isn't super intrusive. 

Maybe some kind of like, hot streak thing? Show recently completed tasks and have it change the color of the parent task accordingly?

**Comment by joshpalan** [comment]
So to start with this, I was thinking we could use the attributes machined_total and actual_total to decide if a task is complete or not. This would prevent us from having to add an extra field in the database that is simply a bool for isComplete. The idea would be that a completed task wouldn't have a machined_total because it won't have any children, and would have an actual_total. However, if a task has a machined_total, you know it's at least not completed. If it has a machined_total and an actual_total (even if it isn't the final actual_total) it's in progress.

The only issue I see with this is that if a task doesn't have any children it won't have a machined_total while it's in progress, but will have an actual_total. This would make youngest children appear complete as soon as they're started. Maybe instead of a completed field, we do an active field that takes a timestamp?

Then when we know there is no machined_time and there is an actual_time, we can check if that field is null to know whether or not that task is in progress. The idea being starting the timer adds a current timestamp to that field and stopping it takes the current timestamp and maths to get the amount of time taken, nulls the in progress field, and sets the total_time to the result.
